src/lib.rs

```diff
@@ -54,13 +54,13 @@ pub enum Error {
 
     /// The value provided for configuration could not be deserialized into the proper type.
     ///
-    /// Check that the serialized data is valid JSON and of the correct type.
-    Deserialization(serde_json_core::de::Error),
+    /// Check that the serialized data is valid and of the correct type.
+    Deserialization,
```

Comment:
Is there a way we could have the de::Error and ser::Error part of the SerDe trait and make the Error enum generic over that?

I've encountered trouble when I discarded error information because it makes it really hard to backtrace where things went wrong, and in this case, I think it traces back all the way over the MQTT python interface to the user, so we should maintain as much information about the error as possible to aid the user in fixing it.

Reply:
Yes. As mentioned that's a downside.
I tried and it didn't look very manageable. Might give it a nother spin.

Reply:
If it looks like a lot of work, let's spawn an issue out to handle it later then. Definitely something that needs fixing